Tests/FluentAssertions.Specs/Xml/XDocumentAssertionSpecs.cs

```diff
@@ -887,8 +886,8 @@ public void When_asserting_a_document_has_a_null_root_element_it_should_fail()
             Action act = () => theDocument.Should().HaveRoot(null);
 
             // Assert
-            act.Should().Throw<ArgumentNullException>().WithMessage(
-                "Cannot assert the document has a root element if the element name is <null>*");
+            act.Should().ThrowExactly<ArgumentNullException>().WithMessage(
```

Comment:
❓ Why was that necessary? Do we care?

Reply:
In general, I try to be as specific as possible when testing for exceptions, so I try to always use `ThrowExactly`.
So, no `AgrumentException` for instance, which would also be fine with `Throw`. But in the end there is currently no difference as long as .NET does not introduce an inherited Exception that inherits from `ArgumentNullException`.

I can revert this change, my point was to have clearer messages, and this was collateral.